math/armstrong_number.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file
+ * \brief A C++ program to check whether a number is armstrong number or not.
+ * 
+ * \details
+ * Armstrong number or [Narcissistic number](https://en.wikipedia.org/wiki/Narcissistic_number) 
+ * is a number that is the sum of its own digits raised to the power of the number of digits.
+ * @author iamnambiar
+*/
+#include<cassert>
+#include<iostream>
+#include<cmath>
+
+/**
+ * Function to calculate the total number of digits in the number.
+ * @param num Number
+ * @return Total number of digits.
+ */
+int number_of_digits(int num) {
+    int total_digits = 0;
+    while (num > 0) {
+        num = num / 10;
+        ++total_digits;
+    }
+    return total_digits;
+}
+
+/**
+ * Function to check whether the number is armstrong number or not.
+ * @param num Number
+ * @return `true` if the number is armstrong.
+ * @return `false` if the number is not armstrong.
+ */
+bool is_armstrong(int number) {
+    // If the number is less than 0, then it is not a armstrong number.
+    if (number < 0) {
+        return false;
+    }
+    int sum = 0;
+    int temp = number;
+    // Finding the total number of digits in the number
+    int total_digits = number_of_digits(number);
+    while (temp > 0) {
+        int rem = temp % 10;
+        // Finding each digit raised to the power total digit and add it to the total sum
+        sum = sum + pow(rem, total_digits);
```

Comment:
```suggestion
        sum = sum + std::pow(rem, total_digits);
```

Reply:
THe pow() method I used is from cmath, does that make any difference in speed?

Reply:
What is the difference here?

Reply:
Using `std::pow` Ensures that the linker uses the correct function implementation from the STL.